Modify Docker (#123)
aidudezzz/deepbots: docker/README.md

# Using deepbots with Docker
## Pull the existing image from [DockerHub](https://hub.docker.com/u/aidudezzz)
* Docker image without conda:
   ```bash
   $ docker pull aidudezzz/deepbots:0.1.3.dev3
   ```
   * You could replace `0.1.3.dev3` with another available tag: https://hub.docker.com/r/aidudezzz/deepbots/tags
* Docker image without conda:
   ```bash
   $ docker pull aidudezzz/deepbots-conda:0.1.3.dev3
   ```
   * You could replace `0.1.3.dev3` with another available tag: https://hub.docker.com/r/aidudezzz/deepbots-conda/tags
## Build and tag a Docker image:

> If you wanted, you could replace `aidudezzz/deepbots:0.1.3.dev3` with another name and optionally a tag in the 'name:tag' format.

Reviewer: "You could replace" makes it similar to the previous section.

Also, maybe "add a tag" sounds more complete? Original wording sounds like "optionally *replace* the tag" but it's not that clear.

```suggestion
You could replace `aidudezzz/deepbots:0.1.3.dev3` with another name and optionally add a tag in the 'name:tag' format.
```

# Using deepbots with Docker
## Pull the existing image from [DockerHub](https://hub.docker.com/u/aidudezzz)
* Docker image without conda:
   ```bash
   docker pull aidudezzz/deepbots:latest
   ```
   * You could replace `latest` with another available tag: https://hub.docker.com/r/aidudezzz/deepbots/tags
* Docker image without conda:
   ```bash
   docker pull aidudezzz/deepbots-conda:latest
   ```
   * You could replace `latest` with another available tag: https://hub.docker.com/r/aidudezzz/deepbots-conda/tags
## Build and tag a Docker image:
You could replace `aidudezzz/deepbots:latest` with another name and optionally add a tag in the 'name:tag' format.
* Docker image without conda:
   ```bash
   docker build -t aidudezzz/deepbots:latest --build-arg branch=dev .
   ```
* Docker image with conda:
   ```bash
   docker build -t aidudezzz/deepbots-conda:latest . --build-arg branch=dev --build-arg conda==true
   ```
In case you would like to add other functionalities/libraries on the Docker image:
* Edit the Dockerfile
* Build the Docker image
* Building arguments
   * `branch`
      * `main`: Install [the latest version of deepbots on PyPI](https://pypi.org/project/deepbots/).

>       * `deepbots==<version>`: Install the specific version of deepbots on [PyPI](https://pypi.org/project/deepbots/) or [TestPyPI](https://test.pypi.org/search/?q=deepbots).

Reviewer: test pypi stuff can also be removed from here.

Author: ```suggestion
      * `deepbots==<version>`: Install the specific version of deepbots on [PyPI](https://pypi.org/project/deepbots/).
```

> 

Author: @tsampazk Yes, that is better! Thank you.

* Edit the Dockerfile
* Build the docker image using below commands
* Building arguments
   * `branch`
      * `dev`: Install [the latest version of deepbots on TestPyPI](https://test.pypi.org/search/?q=deepbots).
      * `main`: Install [the latest version of deepbots on PyPI](https://pypi.org/project/deepbots/).
      * `github`: Install from [the dev branch of deepbots on GitHub](https://github.com/aidudezzz/deepbots/tree/dev).
   * `conda`
      * `true`: Install miniconda.

### Building and tagging a Docker image:
* Docker image without conda:
   ```bash
   $ docker build -t <user-name>/<repo-name> --build-arg branch=dev .
   ```
* Docker image without conda:
   ```bash
   $ docker build -t <user-name>/<repo-name> . --build-arg branch=dev --build-arg conda==true

>    ```

Reviewer: As i don't have a lot of experience with docker i'm a little confused by this. Specifically with `<user-name>/<repo-name>`, at what does this refer to? Could we maybe provide an example?

Author: `-t`: Name and optionally a tag in the 'name:tag' format
* ref: https://docs.docker.com/engine/reference/commandline/build/#options

Actually, users can name their image as `<image-name>`  instead of `<user-name>/<repo-name>`.
`<user-name>/<repo-name>` is for dockerhub users.
I will update it to make it clear. 😄

## Pull the existing image from [DockerHub](https://hub.docker.com/u/aidudezzz)
* Docker image without conda:
   ```bash

>    $ docker pull aidudezzz/deepbots:<tag>

Reviewer: Similar to the previous comment, do you think could we provide an example of a complete link including a tag, so it is really clear to what the user should do?

Author: Thanks!

   ```

>    * tag: https://hub.docker.com/r/aidudezzz/deepbots

Reviewer: I think you meant to link to the tags tab:
```suggestion
   * available tags: https://hub.docker.com/r/aidudezzz/deepbots/tags
```
Also instead of tag->available tags might sound better

Author: Thanks!

* Docker image without conda:
   ```bash
   $ docker pull aidudezzz/deepbots-conda:<tag>
   ```

>    * tag: https://hub.docker.com/r/aidudezzz/deepbots-conda/tags

Reviewer: ```suggestion
   * available tags: https://hub.docker.com/r/aidudezzz/deepbots-conda/tags
```
